### PR TITLE
Override GeldigheidMixin to handle overlapping dates.

### LINKED ISSUE #994: All types should be allowed to exists with the same unique set IFF the others are archived.
**General information**
Open Zaak version: 1.4.0

**Describe the bug**
Similar to #863 but holds for resultaattypen (and prolly more types).

### PR DESCRIPTION
**Pull request by dariomory**
Fixes #994

Duplicating the get_overlapping_x.. function is not in line with DRY principles and ideally it should be transferable to all models.

Proposed solutions:
- A. Override GeldigheidMixin.
- B. Create a new mixing to handle the overlapping dates.

**Changes**
Going with solution A, the GeldigheidMixin has been updated to handle overlapping dates for any models using it.

### CODE REVIEW

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
I believe the duplication/function approach exists because these functions are also called in the API endpoints.

We care more about the admin and API endpoint having the same behaviour than sticking to  DRY behaviour.

Is there a way we can achieve both?

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Comment by dariomory** [comment]
> I believe the duplication/function approach exists because these functions are also called in the API endpoints.
> 
> We care more about the admin and API endpoint having the same behaviour than sticking to DRY behaviour.
> 
> Is there a way we can achieve both?

We can achieve both, at least within DRF's boundaries.

> `verbose_name` is something that is translatable, so this is not a robust check. This logic is also rather "guessing" the field, and that's not something we do in python.
> 
> Please make the field name expliciet by using a class attribute instead, with a sane default on the mixin.

Agree. This was only need due to inconsistent naming of some models fields. What's your opinion on renaming the model field name? That will facilitate the implementation of a DRY function, usable in a single mixin and single validator.

File: src/openzaak/components/catalogi/models/mixins.py [hunk 1]
```diff
@@ -82,6 +90,23 @@ def _clean_geldigheid(self, zaaktype):
                     )
                 )
 
+    def _get_overlapping_objects(self) -> QuerySet:
+        omschrijving_field_name = [i for i in self._meta.fields if i.verbose_name == "omschrijving"][0].name
```
[the remark this answers is not in the document]
Comment (dariomory) [reply src/openzaak/components/catalogi/models/mixins.py:94]: Discussed in https://github.com/open-zaak/open-zaak/pull/995#issuecomment-870608142

**Comment by sergei-maertens** [comment]
@dariomory sorry to get back so late to this

> We can achieve both, at least within DRF's boundaries.

Please do!

> Agree. This was only need due to inconsistent naming of some models fields. What's your opinion on renaming the model field name? That will facilitate the implementation of a DRY function, usable in a single mixin and single validator.

No, do not rename the model fields. That would require us to apply the mapping in the serializers and potentially update this in a lot of places, for something that is very easily solved with a class attribute.

Additionally, I see code style changes where my code suggestion/required changes are being ignored and the conversation is resolved. That's not okay - use the correct black formatting. Black is included in the dev requirements, so you should be using the black version from your virtualenv.

**Review by sergei-maertens — APPROVED** [review]
Elegant!

**Pull request merged**